src/borg/logger.py

```diff
@@ -172,6 +175,7 @@ def info(self, *args, **kw):
             return self.__logger.info(*args, **kw)
 
         def warning(self, *args, **kw):
+            warning_printed = True
             return self.__logger.warning(*args, **kw)
```

Comment:
We're a bit "abusive" in using `warning` and it's often used just for important (should see by default) messages, but those shouldn't change the return code.

Create a new method that does this, say something like `ẃarn[ing]_with_rc` that does this, and leave warning to be just the "important message" thing?